pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -2041,8 +2039,7 @@ impl Timeline {
                     .finish_with_discard_fn(self, ctx, Key::MAX, discard)
                     .await?
             } else {
-                let (layers, _) = writer.take()?;
-                assert!(layers.is_empty(), "image layers produced in dry run mode?");
+                writer.discard();
```

Comment:
could you use `mem::drop`? This function makes it harder to understand what is done here.